docs/reference/builder.md

```diff
@@ -25,31 +25,36 @@ Dockerfile knowledge with the [Dockerfile tutorial](/userguide/level1).
 
 ## Usage
 
-To [*build*](/reference/commandline/cli/#build) an image from a source repository,
-create a description file called `Dockerfile` at the root of your repository.
-This file will describe the steps to assemble the image.
+The `Dockerfile` is used by
```

Comment:
@MHBauer Thank you for the contribution.  This page is needs a few updates so glad you picked it to work on.  Much of the language can be tightened.  It has a lot of notes and warnings indicating structural issues.  

---> https://gist.github.com/moxiegirl/8f1fe9938ce2f85c984f

# Dockerfile reference

Docker can build images automatically by reading the instructions from a
`Dockerfile`. A `Dockerfile` is a text document that contains all the commands a
user could call on the command line to assemble an image. Using `docker build`
users can create an automated build that executes several command-line
instructions in successesion.

This page describes the commands you can use in a `Dockerfile`.  When you are
done reading this page, refer to the [`Dockerfile` Best
Practices](/articles/dockerfile_best-practices) for a tip-oriented guide.

## Usage

The [`docker build`](/reference/commandline/cli/#build) command builds an image from a `Dockerfile` and a *context*. The build's context is the files at a specified location `PATH` or `URL`.  The `PATH` is a directory on your local filesystem.  The `URL` is a the location of a Git repository.

A context is processed recursively. So, a `PATH` includes any subdirectories and the `URL` includes the repository and its submodules.  A simple build command that uses the current directory as context:

    $ docker build .
    Sending build context to Docker daemon  6.51 MB
    Sending build context to Docker daemon 
    ...

The build is run by the Docker daemon, not by the CLI. The first thing a build process does is send the entire context (recursively) to the daemon.  In most cases, it's best to start with empty directory as context and keep
your Dockerfile in that directory. Add only the files needed for
building the Dockerfile.

>**Warning**: Do not use your root directory, `/`, as the `PATH` as it causes
>the build to transfer the entire contents of your hard drive to the Docker
>daemon. 

To use a file in the build context, the `Dockerfile` can refer to it with
command, for example,  a `COPY` instruction. To increase the build's
performance, exclude files and directories by adding a `.dockerignore` file to
the context directory.  For information about how to [create a `.dockerignore`
file](#dockerignore-file) on this page.

Traditionally, the `Dockerfile` is called `Dockerfile` and located in the root of the context. This isn't a requirement though.  You use the `-f` flag with `docker build` to point to a file anywhere in your file sytem.

You can specify a repository and tag at which to save the new image if
the build succeeds: